### DISCUSSION TITLE
`/api/event` throws 500 for self-hosted instance

### DISCUSSION THREAD

**Discussion by gowthamgts** [plausible/analytics#1570] [Self-Hosted Support] [answered]
### Past Issues Searched

- [X] I have searched open and closed issues to make sure that the bug has not yet been reported

### Issue is a Bug Report

- [X] This is a bug report and not a feature request, nor asking for self-hosted support

### Describe the bug

Hello,

I'm trying to use server side event submission via rest api and while trying with `plausible.io/api/event` endpoint it works fine for events with custom properties, whereas in the self-hosted option it seems to be throwing 500.

Sample payload:
```
{
	"name": "download",
	"url": "https://example.com/test1",
	"domain": "example.com",
	"props": {
		"document": "a.txt"
	}
}
```

this works well on plausible.io but not on self-hosted instance. I'm getting the following in server logs:
```
Request: POST /api/event
** (exit) an exception was raised:
    ** (ArgumentError) errors were found at the given arguments:

  * 1st argument: not an iodata term

        :erlang.iolist_to_binary(%{"document" => "a.txt"})
        (jason 1.2.2) lib/jason.ex:55: Jason.decode/2
        (plausible 0.0.1) lib/plausible_web/controllers/api/external_controller.ex:159: PlausibleWeb.Api.ExternalController.parse_meta/1
        (plausible 0.0.1) lib/plausible_web/controllers/api/external_controller.ex:76: PlausibleWeb.Api.ExternalController.create_event/2
        (plausible 0.0.1) lib/plausible_web/controllers/api/external_controller.ex:9: PlausibleWeb.Api.ExternalController.event/2
        (plausible 0.0.1) lib/plausible_web/controllers/api/external_controller.ex:1: PlausibleWeb.Api.ExternalController.action/2
        (plausible 0.0.1) lib/plausible_web/controllers/api/external_controller.ex:1: PlausibleWeb.Api.ExternalController.phoenix_controller_pipeline/2
        (phoenix 1.5.13) lib/phoenix/router.ex:352: Phoenix.Router.__call__/2
```

this happens only when i have `props` in the payload, else it's working fine.

### Expected behavior

should get 202

### Environment

```markdown
- OS: Ubuntu 20.04
- Browser: Tested via insomnia rest client
- Browser Version: NA
```

**Comment by ukutaht** [comment, marked as answer]
This was fixed in https://github.com/plausible/analytics/commit/7a02aae2a562efd39f11fa405c0f084c4d59e8cc

Previously we expected the props to be JSON-stringified. This means double-encoding like this:

```json
{
	"name": "download",
	"url": "https://example.com/test1",
	"domain": "example.com",
	"props": "{\"document\": \"a.txt\"}"
}
```

This is quite ugly and suboptimal so I made a change where you can send them as part of the regular JSON payload. The fix is not released yet but in the meanwhile you can stringify the props before sending them.

**Reply by gowthamgts** [reply]
thanks @ukutaht - this indeed works. just for preparation on my side, when will these new build land on `plausible/analytics` docker image?

**Reply by ukutaht** [reply]
We cut releases every 6 months or so. Sorry it's not very frequent. We just released 1.4 so the next one will be this spring / early summer

**Reply by gowthamgts** [reply]
Sure thanks
